e2e/utils/wait-utils.ts

```diff
@@ -0,0 +1,137 @@
+import {Page} from 'puppeteer';
+
+/**
+ * Summary: Helper functions when directly dealing with XPath selectors.
+ */
+
+export async function waitForProperty(page: Page, xpathSelector: string, propertyName: string, propertyValue: string, maxTimeOut: number = 30000) {
+  return page.waitForFunction(xpath => {
+    const element = document.evaluate(
```

Comment:
Is there another way to call this method without so many parameters (like with an options struct)?